15-components-lab_v16/src/00-synctext.jsx

```diff
@@ -1,35 +1,44 @@
 import React from 'react';
-import PropTypes from 'prop-types';
 
-class SingleInput extends React.Component{
-  static propTypes = {
-    val: PropTypes.string
+export default class SingleInput extends React.Component{
+
+  constructor(props){
+    super(props);
+    this.state= { val : 'type something...'}
   }
 
-  static defaultProps = {
-    val: 'type something...'
+
+  onChange = (e) => {

```

Comment:
Did you try `this.setState(() => ({ ... }))` (the functional form)? Did it work? Can you explain why not?